release_util/management/commands/__init__.py

```diff
@@ -131,165 +134,241 @@ def state(self):
-                         stdout=out,
-                         database=self.database_name)
-        except (CommandError, DatabaseError) as e:
-            time_to_fail = self.timer() - start
-            exc_type, exc_value, exc_traceback = sys.exc_info()
+    def list_migrations(self):
+        """
+        Returns a tuple of unapplied, current
 
-            # Parse the output to discover the last migration being applied before the exception.
-            # Mark migrations:
-            # - before exception migration as success
-            # - exception migration as failed
-            # - any remaining exceptions as unapplied
-            for line in out.getvalue().split('\n'):
-                line = _remove_escape_characters(line).strip()
-                line_match = self.migration_regex.match(line)
-                if line_match:
-                    migration = [line_match.group('app_name'), line_match.group('migration_name')]
-                    if line_match.group('success') == 'OK':
-                        # A migration succeeded.
-                        self.migration_state['success'].append({
-                            'migration': migration,
-                            'output': line,
-                        })
-                    else:
-                        # This migration failed.
-                        failed_app, failed_migration = migration
-                        self.migration_state['failure'] = {
-                            'migration': migration,
-                            'output': out.getvalue(),
-                            'traceback': repr(traceback.format_exception(exc_type, exc_value, exc_traceback)),
-                        }
-                        # Save the total migration time until failure.
-                        self.migration_state['duration'] = time_to_fail
-                        break
-
-            # Remove any migrations that succeeded -or- failed.
-            # Any leftover migrations were *not* applied.
-            succeeded = [status['migration'] for status in self.migration_state['success']]
-            failed = []
-            if failed_app and failed_migration:
-                failed = [failed_app, failed_migration]
-            for migration in self.to_apply:
-                if migration not in succeeded and migration != failed:
-                    self.migration_state['unapplied'].append(migration)
+        "Unapplied" is a list of unapplied migrations. "Current" is a list of the current migration
+        states for apps with unapplied migrations.
 
-            # Find the apps that have been applied -or- failed.
-            # Include their initial migrations as commands.
-            self._add_rollback_commands()
-            raise CommandError("Migration failed for app '{}' - migration '{}'.\n".format(failed_app, failed_migration))
+        Both are tuples of the form (app, migration_name).
+        """
+        connection = connections[self._database_name]
```

Comment:
nit: I do not see connection used elsewhere (I assume because we do the migration via the django command) but connection strikes me as something that should be a class member.